src/platform/Darwin/DeviceInstanceInfoProviderImpl.cpp

```diff
@@ -23,15 +23,88 @@
 namespace chip {
 namespace DeviceLayer {
 
+CHIP_ERROR DeviceInstanceInfoProviderImpl::GetVendorName(char * buf, size_t bufSize)
+{
+    CHIP_ERROR err;
+    size_t vendorNameLen = 0; // without counting null-terminator
+    err = Internal::PosixConfig::ReadConfigValueStr(Internal::PosixConfig::kConfigKey_VendorName, buf, bufSize, vendorNameLen);
```

Comment:
ReadConfigValueStr does not null-terminate the buffer, so this code is a security bug as written.